### DISCUSSION TITLE
Why does the upload endpoint already rate-limit after 6 Next.js deployments?  (5000 Uploads per day (Free))

### DISCUSSION THREAD

**Discussion by yannxaver** [vercel/community#5115] [Help] [answered]
### Summary

I use Github Actions to build my next.js application and then deploy it to Vercel prebuilt. After only 6 deployments today, I already got rate-limited by the uploads endpoint. Uploads are supposed to be limited to 5000 per day.

`DeploymentError: Too many requests - try again in 18 hours (more than 5000, code: "api-upload-free").`

Do six deployments of a basic prebuilt next.js application use up that limit already? Building and deploying via github actions doesn't seem feasible then at all.

### Steps to Reproduce

/

**Comment by swarnava** [comment, marked as answer]
Hi,

 This limit is the amount of files you're able to upload via the API/CLI in a 24 hour period. Typically users hit this limit when they're uploading a large amount of additional files, such as `node_modules` or Git `metadata`.

In the initial instance, we'd recommend double checking what's been uploaded to Vercel and confirming the files are necessary. 
If you can't reduce the amount of files uploaded, we include an archive feature which archives everything and then only uploads the single archive file. This means each upload takes slightly longer as it needs to archive on your side and extract on Vercel's end, although with a large amount of files it usually ends up being the same speed. 

If you wish to use the archive feature, this can be configured by adding the following command to the end of your deploy command: --archive=tgz (eg 'vercel deploy --archive=tgz')

**Reply by yannxaver** [reply]
> In the initial instance, we'd recommend double checking what's been uploaded to Vercel and confirming the files are necessary.

![Screenshot 2023-12-08 at 17 44 03](https://github.com/vercel/community/assets/17554280/faab9b95-f4c0-4b37-a362-50ffa85f6201)
Looks okay, what do you think?

> If you can't reduce the amount of files uploaded, we include an archive feature which archives everything and then only uploads the single archive file.

I will try that out once my rate limit has expired tomorrow. If you could lift it before that, that would be great too.

> If you wish to use the archive feature, this can be configured by adding the following command to the end of your deploy command: --archive=tgz (eg 'vercel deploy --archive=tgz')

That sounds like a good solution, I will update the Github action. By the way, I have basically copy-pasted the example from the website (https://vercel.com/guides/how-can-i-use-github-actions-with-vercel). Could be a good idea to update the deploy command in there as well.

**Reply by swarnava** [reply]
'vercel deploy --archive=tgz` is an experimental feature so we don't document it publicly. Any specific reason why you aren't using native github integration? You can also use `vercel deploy` instead of building it using `vercel build` without `--prebuilt`

**Reply by yannxaver** [reply]
I run tests and migrations in the workflow and only trigger the deployment if they completed successfully. 

How would I achieve that using the native git integration? My understanding was that it triggers a deployment on every commit - no matter what.

I will also look into just running `vercel deploy` instead and deferring the build to Vercel. I was assuming that it would upload the entire source code every time which would lead to similar rate limiting.
